### PR TITLE
chore: add minimum node version

### PR DESCRIPTION
**Pull request by eunjae-lee**
This PR adds a minimum node version to `package.json`.

### CODE REVIEW

**Comment by eunjae-lee** [comment]
This is based on `develop` which means this branch is not ready yet for node module.
So in order to test this, what I did is

- comment out these two sentences from `insights.ts`
  - `this.setUserToken(this.ANONYMOUS_USER_TOKEN);`
  - `this.processQueue(window);`
- `yarn link`
- `cd` to an empty project
- `nvm install 8.15.1 && nvm use 8.15.1`
- `yarn link search-insights && yarn add search-insights`

With `engineStrict`, I was expecting to see an error, but I didn't.
It was same with `npm link`.
I'm not sure what I missed.

File: package.json [hunk 1]
```diff
@@ -4,6 +4,10 @@
   "description": "Search analytics by Algolia",
   "jsdelivr": "dist/search-insights.min.js",
   "main": "dist/search-insights.min.js",
+  "engineStrict": true,
```
Comment (Haroenv) [inline package.json]: where's the documentation of this? I haven't seen it before

**Comment by Haroenv** [comment]
@eunjae-lee, what if you do `yarn pack`, and do `yarn install ../search-insights.js/search-insights-XXX.tar.gz` in the other repository? I've definitely seen errors like that before

File: package.json (hunk 1 above)
Comment (eunjae-lee) [reply package.json]: Oh, now I checked again and it's removed a long time ago. I'm removing this.

**Comment by Haroenv** [comment]
`engineStrict` isn't a thing by the way: https://npm.community/t/engines-and-engines-strict-ignored/4792/2

**Comment by eunjae-lee** [comment]
> @eunjae-lee, what if you do `yarn pack`, and do `yarn install ../search-insights.js/search-insights-XXX.tar.gz` in the other repository? I've definitely seen errors like that before

ah okay I will test it that way. thanks

**Comment by eunjae-lee** [comment]
Okay, with `yarn pack`, it works fine!

File: package.json (hunk 1 above)
Comment (eunjae-lee) [reply package.json]: https://github.com/algolia/search-insights.js/pull/146/commits/553aec6a12047c46f899dae04d6d0398a41ae4fa

**Pull request merged**